### PR TITLE
fix(blink): auto friend peers to aid in discovery

### PR DESCRIPTION
**Pull request by sdwoodbury**
**What this PR does** 📖
sometimes testing the `blink-cli` is hampered by not being able to discover each other. `Multipass` has a built in discovery mechanism. This PR attempts to make use of that by auto-friending anyone dialed using the `blink-cli`. 

**Which issue(s) this PR fixes** 🔨

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Comment by tooshel** [comment]
👍

File: tools/blink-cli/src/main.rs [hunk 1]
```diff
@@ -119,6 +122,7 @@ enum Repl {
 
 async fn handle_command(
     blink: &mut Box<dyn Blink>,
+    multipass: &Arc<Mutex<Box<dyn MultiPass>>>,
```
Comment (dariusc93) [inline tools/blink-cli/src/main.rs:125]:
```suggestion
    multipass: &Box<dyn MultiPass>,
```

File: tools/blink-cli/src/main.rs [hunk 2]
```diff
@@ -278,6 +283,19 @@ async fn handle_event_stream(mut stream: BlinkEventStream) -> anyhow::Result<()>
     Ok(())
 }
 
+async fn handle_multipass_event_stream(
+    multipass: Arc<Mutex<Box<dyn MultiPass>>>,
```
Comment (dariusc93) [inline tools/blink-cli/src/main.rs:287]:
```suggestion
    multipass: Box<dyn MultiPass>,
```

File: tools/blink-cli/src/main.rs [hunk 3]
```diff
@@ -278,6 +283,19 @@ async fn handle_event_stream(mut stream: BlinkEventStream) -> anyhow::Result<()>
...
+    mut stream: MultiPassEventStream,
+) -> anyhow::Result<()> {
+    while let Some(evt) = stream.next().await {
+        if let MultiPassEventKind::FriendRequestReceived { from } = evt {
+            let _ = multipass.lock().await.accept_request(&from).await;
```
Comment (dariusc93) [inline tools/blink-cli/src/main.rs:292]:
```suggestion
            let _ = multipass.accept_request(&from).await;
```

File: tools/blink-cli/src/main.rs [hunk 4]
```diff
@@ -338,11 +356,20 @@ async fn main() -> anyhow::Result<()> {
         }
     };
 
-    let mut blink: Box<dyn Blink> = warp_blink_wrtc::BlinkImpl::new(multipass).await?;
-    let event_stream = blink.get_event_stream().await?;
-    let handle = tokio::spawn(async {
-        if let Err(e) = handle_event_stream(event_stream).await {
-            println!("handle event stream failed: {e}");
+    let mut blink: Box<dyn Blink> = warp_blink_wrtc::BlinkImpl::new(multipass.clone()).await?;
+    let multipass = Arc::new(Mutex::new(multipass));
```
Comment (dariusc93) [inline tools/blink-cli/src/main.rs:360]:
```suggestion
```

File: tools/blink-cli/src/main.rs [hunk 5]
```diff
@@ -338,11 +356,20 @@ async fn main() -> anyhow::Result<()> {
...
+    let blink_event_stream = blink.get_event_stream().await?;
+    let blink_handle = tokio::spawn(async {
+        if let Err(e) = handle_blink_event_stream(blink_event_stream).await {
+            println!("handle blink event stream failed: {e}");
+        }
+    });
+
+    let multipass_event_stream = multipass.lock().await.subscribe().await?;
```
Comment (dariusc93) [inline tools/blink-cli/src/main.rs:368]:
```suggestion
    let multipass_event_stream = multipass.subscribe().await?;
```

**Review by dariusc93 — COMMENTED** [review]
Would have to check the IDE for any errors, but unless there is a specific use case, there shouldnt be any need to place the boxed trait object into a arc or a lock primitive due to the internals being thread safe. 

Also, using `MultiPass::get_identity` should trigger an discovery attempt (since internally its adds the public key and attempts to search or connect based on the configuration).

**Review by dariusc93 — APPROVED** [review]

**Pull request merged**